core/domain/feedback_jobs_one_off.py

```diff
@@ -182,3 +184,95 @@ def map(item):
+            # else it has already been migrated.
+            if len(old_id.split('.')) == 2:
+                new_id = 'exploration.' + item.id
+                feedback_models.FeedbackThreadModel(
+                    id=new_id, entity_type=item.entity_type,
+                    entity_id=item.entity_id,
+                    exploration_id=item.exploration_id,
+                    state_name=item.state_name,
+                    original_author_id=item.original_author_id,
+                    status=item.status, subject=item.subject,
+                    summary=item.summary, has_suggestion=item.has_suggestion,
+                    message_count=item.message_count,
+                    last_updated=item.last_updated, created_on=item.created_on,
+                    deleted=item.deleted).put()
+        elif isinstance(item, feedback_models.FeedbackMessageModel):
+            old_id = item.id
+            if len(old_id.split('.')) == 3:
+                new_id = 'exploration.' + item.id
+                new_thread_id = 'exploration.' + item.thread_id
+                feedback_models.FeedbackMessageModel(
+                    id=new_id, thread_id=new_thread_id,
+                    message_id=item.message_id, author_id=item.author_id,
+                    updated_status=item.updated_status,
+                    updated_subject=item.updated_subject, text=item.text,
+                    received_via_email=item.received_via_email,
+                    last_updated=item.last_updated, created_on=item.created_on,
+                    deleted=item.deleted).put()
+        elif isinstance(item, feedback_models.FeedbackThreadUserModel):
+            old_id = item.id
+            if len(old_id.split('.')) == 3:
+                new_id = '.'.join(
+                    [old_id.split('.')[0], 'exploration', old_id.split('.')[1],
+                     old_id.split('.')[2]])
+                feedback_models.FeedbackThreadUserModel(
+                    id=new_id,
+                    message_ids_read_by_user=item.message_ids_read_by_user
+                ).put()
+        elif isinstance(item, email_models.FeedbackEmailReplyToIdModel):
+            old_id = item.id
+            if len(old_id.split('.')) == 3:
+                new_id = '.'.join(
+                    [old_id.split('.')[0], 'exploration', old_id.split('.')[1],
+                     old_id.split('.')[2]])
+                email_models.FeedbackEmailReplyToIdModel(
+                    id=new_id, reply_to_id=item.reply_to_id).put()
+        elif isinstance(item, user_models.UserSubscriptionsModel):
+            for thread_id in item.feedback_thread_ids:
+                if len(thread_id.split('.')) == 2:
+                    new_thread_id = 'exploration.' + thread_id
+                    if new_thread_id not in item.feedback_thread_ids:
+                        item.feedback_thread_ids.append(new_thread_id)
+            item.put()
```

Comment:
Not sure but does this guarantee that unnecessary (when nothing is updated) write do not happen?

Reply:
I don't think this guarantees that (because the last_updated field would change right?). However, as we are doing it differently (using new models), I don't think it is problematic.